Added relative optional padding support

steurt opened the pull request:
Added relative optional padding support and applied some minor code shortening

Commit:
Added relative optional padding support

Commit:
Added useStyledAttributes to work with a TypedArray even simpler combining obtainStyledAttributes and the already available TypedArray.use block

dewarder commented on akommons/src/main/java/com/dewarder/akommons/view/View.kt:
```diff
@@ -60,34 +62,35 @@ inline fun <V : View> View.findView(
-fun View.setAllPadding(padding: Int) {
-    setPadding(padding, padding, padding, padding)
-}
+fun View.setAllPadding(padding: Int) = setPadding(padding, padding, padding, padding)
```
Please, revert back all functions to use block body instead of expression one. The purpose of block body is to prevent accidentally changing of return type and omit explicit `Unit` as return type.

dewarder commented on akommons/src/main/java/com/dewarder/akommons/content/res/TypedArray.kt:
```diff
@@ -37,4 +43,16 @@ inline fun <R> TypedArray.use(block: (TypedArray) -> R): R {
\ No newline at end of file
+}
+
+inline fun <R> Context.useStyledAttributes(@StyleableRes attrs: IntArray, block: (TypedArray) -> R): R
```
It will be better to move all extensions with receiver of `Context` to file with same name which is placed in `akommons.content.Context`

dewarder commented on akommons/src/main/java/com/dewarder/akommons/content/res/TypedArray.kt:
```diff
@@ -37,4 +43,16 @@ inline fun <R> TypedArray.use(block: (TypedArray) -> R): R {
+inline fun <R> Context.useStyledAttributes(@StyleableRes attrs: IntArray, block: (TypedArray) -> R): R
+    = obtainStyledAttributes(attrs).use(block)
+
+inline fun <R> Context.useStyledAttributes(@StyleRes resid: Int, @StyleableRes attrs: IntArray, block: (TypedArray) -> R): R
```
Please, format all parameters of functions according to [Coding Conventions](https://kotlinlang.org/docs/reference/coding-conventions.html#function-formatting) in case if it doesn't fit 80 symbols limit.

dewarder commented on akommons/src/main/java/com/dewarder/akommons/content/res/TypedArray.kt:
```diff
@@ -37,4 +43,16 @@ inline fun <R> TypedArray.use(block: (TypedArray) -> R): R {
+inline fun <R> Context.useStyledAttributes(set: AttributeSet, @StyleableRes attrs: IntArray, block: (TypedArray) -> R): R
+    = obtainStyledAttributes(set, attrs).use(block)
+
+inline fun <R> Context.useStyledAttributes(set: AttributeSet, @StyleableRes attrs: IntArray, @AttrRes defStyleAttr: Int, @StyleRes defStyleRes: Int, block: (TypedArray) -> R): R
```
Please, add zeros as default parameters to `defStyleAttr` and `defStyleRes`. It will make redundant function above (with signature `Context.useStyledAttributes(set, attrs, block)`)

dewarder commented:
Hello @TymenSteur . Thanks a lot for your contribution! I will be glad to merge your changes after comments will be fixed.

Commit:
Moved useStyledAttributes extensions to the proper reciever file Context and updated some code to match conventions

TymenSteur commented:
Great, thanks for accepting the additions!
Any idea when I can expect a new release including the changes?
